src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java

```diff
@@ -63,16 +65,33 @@ public void execute(IFileDeployCommandData context) {
                     }
                     FilePath file = zipFilesPathList.get(0);
                     context.logStatus("Deploy to app " + file.getBaseName() + " using file: " + file.getRemote());
-                    try (InputStream stream = file.read()) {
-                        String slotName = context.getSlotName();
-                        if (StringUtils.isEmpty(slotName)) {
-                            app.zipDeploy(stream);
-                        } else {
-                            DeploymentSlot slot = app.deploymentSlots().getByName(slotName);
-                            slot.zipDeploy(stream);
+
+                    int retryCount = 0;
+                    while (retryCount < DEFAULT_MAX_RETRY_TIMES) {
+                        retryCount++;
+                        try (InputStream stream = file.read()) {
+                            String slotName = context.getSlotName();
+                            try {
+                                if (StringUtils.isEmpty(slotName)) {
+                                    app.zipDeploy(stream);
+                                } else {
+                                    DeploymentSlot slot = app.deploymentSlots().getByName(slotName);
+                                    slot.zipDeploy(stream);
+                                }
+                                context.setCommandState(CommandState.Success);
+                                AzureAppServicePlugin.sendEvent(Constants.AI_WEB_APP, aiDeployInfo,
+                                        "Run", AppInsightsUtils.hash(context.getJobContext().getRun().getUrl()),
+                                        "ResourceGroup", AppInsightsUtils.hash(context.getWebApp().resourceGroupName()),
+                                        "WebApp", AppInsightsUtils.hash(context.getWebApp().name()));
+                                return;
+                            } catch (Exception e) {
+                                context.logStatus(String.format("Exception occurred when deploying "
+                                                + "the zip package: %s, retrying immediately (%d/%d)", e.getMessage(),
+                                        retryCount, DEFAULT_MAX_RETRY_TIMES));
+                            }
                         }
                     }
-                    break;
+                    throw new IOException(String.format("The zip deploy failed after %d times of retry.", retryCount));
```

Comment:
IOException? is there any other more appriopriate exception type?

Reply:
fixed